presto-main/src/main/java/com/facebook/presto/type/TypeRegistry.java

```diff
@@ -458,16 +504,120 @@ else if (isCovariantParameterPosition(commonSuperTypeBase.get(), i)) {
         return Optional.of(new TypeSignature(commonSuperTypeBase.get(), typeParameters.build()));
     }
 
-    /*
-     * Given signature with unmatched parameters return filtered one.
+    /**
+     * In case of coercion between un-parametrized types (DOUBLE, BIGINT, ...)
+     * we only need to know whether coercion is allowed (true|false).
+     * But in case of coercion to the type with long parameters, parameters of the result type must be calculated.
+     * Such coercion rules must be implemented within this method.
+     *
+     * Examples:
+     *
+     * LONG, DECIMAL(x,y) -> DECIMAL(19,0)
```

Comment:
What cases rely on this? I find the concept of looking for the common type of a concrete type and an unbound parametric type a bit strange for what the type system can do today. If anything, the result should be another unbound type with additional constraints, but that doesn't seem very useful. E.g.,

`LONG` vs `DECIMAL(x,y)` -> `DECIMAL(a,b)`, with  `a = max(x, 19)`, `b = max(y, 0)`